src/metabase/driver/sql_jdbc/sync.clj

```diff
@@ -96,12 +96,14 @@
 (defn- db-tables
   "Fetch a JDBC Metadata ResultSet of tables in the DB, optionally limited to ones belonging to a given
   schema."
-  [^DatabaseMetaData metadata ^String schema-or-nil ^String db-name-or-nil]
+  [driver ^DatabaseMetaData metadata ^String schema-or-nil ^String db-name-or-nil]
   ;; tablePattern "%" = match all tables
-  (with-open [rs (.getTables metadata db-name-or-nil schema-or-nil "%"
+  (with-open [rs (.getTables metadata db-name-or-nil (driver/metadata-escape-entity-name driver schema-or-nil) "%"
                              (into-array String ["TABLE" "VIEW" "FOREIGN TABLE" "MATERIALIZED VIEW" "EXTERNAL TABLE"]))]
-    (mapv #(select-keys % [:table_name :remarks :table_schem])
-          (jdbc/result-set-seq rs))))
+    (transduce (map (fn [row] (select-keys row [:table_name :remarks :table_schem])))
```

Comment:
This could be even more concisely written using `into` (while retaining transducer semantics):
```
(into [] 
  (map (fn [row] (select-keys row [:table_name :remarks :table_schem]))) 
  (jdbc/reducible-result-set rs {}))
```